Not compatible with React 17

Hi,
The[ peer dependencies on react](https://github.com/craig1123/react-recipes/blob/master/package.json#L67) prevents us from using it with react 17 (the default version used by nextjs 10 for example, and probably the latest CRA version...).

```
npm ERR! While resolving: myApp@0.1.0
npm ERR! Found: react@17.0.1
npm ERR! node_modules/react
npm ERR!   react@"17.0.1" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.8.0" from react-recipes@1.1.0
npm ERR! node_modules/react-recipes
npm ERR!   react-recipes@"^1.1.0" from the root project
npm ERR!
```

I will open a PR soon to should solve this issue.

--- a/package.json
+++ b/package.json
@@ -64,8 +64,8 @@
     "rimraf": "^3.0.2"
   },
   "peerDependencies": {
-    "react": "^16.8.0",
-    "react-dom": "^16.8.0"
+    "react": "^16.8.0 || ^17.0.0",
+    "react-dom": "^16.8.0 || ^17.0.0"
   },
   "jest": {
     "coverageThreshold": {

